topydo/lib/PrettyPrinterFilter.py

```diff
@@ -57,15 +60,21 @@ def filter(self, p_todo_str, p_todo):
             except KeyError:
                 pass
 
-            p_todo_str = color + p_todo_str + NEUTRAL_COLOR
-
+            p_todo_str = '%s%s%s' % (color, p_todo_str, NEUTRAL_COLOR)
             if config().highlight_projects_contexts():
                 p_todo_str = re.sub(
                     r'\B(\+|@)(\S*\w)',
-                    PROJECT_COLOR + r'\g<0>' + color,
+                    #PROJECT_COLOR + r'\g<0>' + color,
+                    lambda m: (
+                        CONTEXT_COLOR if m.group(0)[0] == "+"
+                        else PROJECT_COLOR)+m.group(0)+color,
                     p_todo_str)
-
-            p_todo_str += NEUTRAL_COLOR
+            p_todo_str = re.sub(r'\w(\S*:[^/\s]\S*)\w',METADATA_COLOR+r'\g<0>'+color,p_todo_str)
+            # add LINK_COLOR to anything like :  (one group od charater and a :)(// and non whitespace chars)
+            p_todo_str = re.sub(r'\s(\w*:){1}(//\S*\w)',' '+LINK_COLOR+r'\1\2'+color,p_todo_str)
```

Comment:
Perhaps rewrite to `\b\S+://\S+\b` ? This matches the case when a word contains `://`. Or did I miss a corner case?